pennylane_lightning/core/src/simulators/lightning_tensor/tncuda/bindings/LTensorTNCudaBindings.hpp

```diff
@@ -53,12 +53,27 @@ using TensorNetBackends =
 template <class TensorNet, class PyClass>
 void registerBackendClassSpecificBindings(PyClass &pyclass) {
     registerGatesForTensorNet<TensorNet>(pyclass);
+    using PrecisionT = typename TensorNet::PrecisionT; // TensorNet's precision
+    using ParamT = PrecisionT; // Parameter's data precision
+
+    using np_arr_c = py::array_t<std::complex<ParamT>,
+                                 py::array::c_style | py::array::forcecast>;
 
     pyclass
         .def(py::init<const std::size_t,
                       const std::size_t>()) // num_qubits, max_bond_dim
         .def(py::init<const std::size_t, const std::size_t,
                       DevTag<int>>()) // num_qubits, max_bond_dim, dev-tag
+        .def(
+            "getState",
```

Comment:
Do we need any checks to catch data overflows in case the destination Numpy buffer is insufficient for the copy?

Reply:
yes, and it's checked inside the `getData()` API.